static/js/views/discovered-addon.js

```diff
@@ -123,6 +129,55 @@ class DiscoveredAddon {
         controlDiv.innerHTML = el;
       });
   }
+  
+  /**
+   * Handle a click on the license button.
+   */
+  handleLicense(e) {
+		if(e.target.getAttribute('data-id')) {
+			const license_url = "https://api.mozilla-iot.org:8443/addons/license/" + e.target.getAttribute('data-id');
+
+			var modal = document.getElementById('media-modal');
+			
+			if(modal == null){
+		  	var modal_container = document.createElement('div');
+		  	modal_container.className = "media-modal";
+				modal_container.id = "media-modal";
+			
+		  	var modal_frame = document.createElement('div');
+		  	modal_frame.className = "media-modal-frame";
+		  	modal_frame.innerHTML = '<div class="media-modal-close" id="modal-close-button"></div><div class="media-modal-content"><p id="media-modal-text">Loading...</p></div>';
+
+		  	modal_container.appendChild(modal_frame);
+		  	document.body.appendChild(modal_container);
+				
+				var modal_close_button = document.getElementById('modal-close-button');
+		    modal_close_button.addEventListener(
+		      'click',
+		      () => {
+					  modal = document.getElementById("media-modal");
+					  modal.parentNode.removeChild(modal);
+		      }
+		    );
+				
+				fetch(license_url).then(response => {
+				  if (!response.ok) {
+				    throw new Error("HTTP error " + response.status);
+						document.getElementById("media-modal-text").innerText = "Error while trying to load license";
+				  }
+				  return response.text();
+				})
+				.then(data => {
+				  document.getElementById("media-modal-text").innerText = data;
+			  })
+				.catch((error) => {
+				  document.getElementById("media-modal-text").innerText = "Connection error while trying to load license";
```

Comment:
String should be moved into translation.

Reply:
Interesting. Will do.